packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -159,6 +164,17 @@ class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
 				assert(knownSessionId.value === value.value, 0xa5a /* Mismatched SessionId */);
 			}
 		}
+
+		const audienceMembers = this.audience.getMembers();
+		const attendees = new Set(this.attendees.values());
```

Comment:
What is the `Set` doing?

Reply:
this.attendees.values() returns a list sessionIds that are associated with several connectionIds. The Set is used here to only get the unique sessionIds.

Reply:
Right. Feels inefficient to visit every attendee every time there is an update. Perhaps you can start with empty set and accumulate those mentioned, that appear connected, to limit processing.

Reply:
New algo for better efficiency: 
- start with empty connectedAttendee set
- for each connectionid mentioned:
       - if connectionid is in audience, add corresponding attendee to connected set and set as "Connected". 
       - if attendee not in connected set, set attendee to "Disconnected".